docs/modules/trino/pages/usage-guide/operations/graceful-shutdown.adoc

```diff
@@ -1,32 +1,65 @@
+2. The worker sleeps for `30` seconds to ensure that the coordinator has noticed the shutdown and stops scheduling new tasks on the worker.
 3. The worker now waits till all tasks running on it complete. This will take as long as the longest running query takes.
-4. The worker sleeps for 60 seconds to ensure that the coordinator has
+4. The worker sleeps for `30` seconds to ensure that the coordinator has
 noticed that all tasks are complete
 5. The `PreStop` hook will never return, but the JVM will be shut down by the graceful shutdown mechanism.
-6. When the graceful shutdown is not quick enough (e.g. a query runs longer than the graceful shutdown period), after `<graceful shutdown period> + 60s of step 2 + 60s of step 4 + 30s safety overhead` the Pod gets killed, regardless if it has shut down gracefully or not. This is achieved by setting `terminationGracePeriodSeconds` on the worker Pods.
+6. When the graceful shutdown is not quick enough (e.g. a query runs longer than the graceful shutdown period), after `<graceful shutdown period> + 30s of step 2 + 30s of step 4 + 10s safety overhead` the Pod gets killed, regardless if it has shut down gracefully or not. This is achieved by setting `terminationGracePeriodSeconds` on the worker Pods. In this case queries running on the worker will fail.
 
-WARNING: As of 23.7, the secret-operator issues TLS certificates with a lifetime of 24h. It also adds an annotation to the Pod, so that it is restarted 30 minutes before the certificate expires (23.5h hours in this case). Bot can not be configured. This results in all Pod using https (both coordinator and workers in a typical setup) restarting every 23.5 hours. This problem will be addressed in a future release by e.g. making the certification lifetime configurable.
+WARNING: As of version 23.7, the secret-operator issues TLS certificates with a lifetime of 24h. It also adds an annotation to the Pod, so that it is restarted 30 minutes before the certificate expires (23.5h hours in this case). Currently, this results in all Pod using https (both coordinator and workers in a typical setup) restarting every 23.5 hours.
+The TLS certificate lifetime can be configured once https://github.com/stackabletech/secret-operator/pull/306 is merged.
 
 == Implications
-All queries that take less than the graceful shutdown period are guaranteed to not be disturbed by regular termination of Pods.
-They can obviously still fail when e.g. a Kubernetes node dies completely or the Pod does not get the time it takes (e.g. 1h by default) to properly gracefully shut down.
+All queries that take less than the minimal graceful shutdown period of all roleGroups (`1` hour as a default) are guaranteed to not be disturbed by regular termination of Pods.
+They can obviously still fail when e.g. a Kubernetes node dies completely or the Pod does not get the time it takes to properly gracefully shut down.
```

Comment:
```suggestion
They can obviously still fail when, for example, a Kubernetes node dies or the graceful shutdown timeout configured on the Pod is not long enough.
```

Reply:
Changed to `They can obviously still fail when, for example, a Kubernetes node dies completely or the Pod does not get the time it takes to properly gracefully shut down - e.g. the Kubernetes node gets rebooted before it is fully drained.`

Reply:
This version is slightly more compact: `They can obviously still fail when, for example, a Kubernetes node dies or gets rebooted before it is fully drained.`

Reply:
Changed in 41596a4